c10/util/BFloat16.h

```diff
@@ -5,21 +5,37 @@
 
 #include <c10/macros/Macros.h>
 #include <cmath>
+#include <cstring>
 
 namespace c10 {
 
 namespace detail {
+  inline bool isSmallEndian() {
+    int num = 1;
+    return *(char *)&num == 1;
+  }
+
   inline float f32_from_bits(uint16_t src) {
     float res = 0;
-    uint16_t* tmp = reinterpret_cast<uint16_t*>(&res);
-    tmp[0] = 0;
-    tmp[1] = src;
+    uint32_t tmp = src;
+    if (isSmallEndian()) {
+      tmp <<= 16;
```

Comment:
did you test this?  This doesn't seem right -- the bitshift operators should do the 'right' thing given the endianness of the system.